staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -410,55 +442,162 @@ func TestReflectorStopWatch(t *testing.T) {
 	ctx, cancel := context.WithCancelCause(ctx)
 	cancel(errors.New("don't run"))
 	err := handleWatch(ctx, time.Now(), fw, s, g.expectedType, g.expectedGVK, g.name, g.typeDescription, g.setLastSyncResourceVersion, g.clock, nevererrc)
+	defer func() {
+		if fw != nil {
+			fw.Stop()
+		}
+	}()
 	if err != errorStopRequested {
 		t.Errorf("expected stop error, got %q", err)
 	}
 }
 
-func TestReflectorListAndWatch(t *testing.T) {
+func TestReflectorListAndWatchWithWatchList(t *testing.T) {
```

Comment:
Can we merge those tests into a single one?

Isn't it that they really only differ with `expectedRVs` and `expectedListRVs` ? 
I would like to avoid code duplication and just share it and parametrize it different in those two cases.

Reply:
Realistically, this whole package of tests should be rewritten to test both branches. Most of them assume the old ListAndWatch behavior. But I can do it for just this one...

Reply:
Done